pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java

```diff
@@ -263,7 +264,14 @@ public void addServlet(String path, ServletHolder servletHolder, boolean require
             });
         }
         filterInitializer.addFilters(context, requiresAuthentication);
-        handlers.add(context);
+        // Enable compress on /metrics endpoint
+        if (path.equals("/metrics") && pulsar.getConfiguration().isCompressOutputMetricsInPrometheus()) {
```

Comment:
This is quite confusing for someone coming later to this part of code. I would expect this to be where the /metrics is defined. 
I prefer adding a boolean argument to `addServlet` asking if this servlet should be GZIpped and if so, then add wrap the servlet context handler with a `GZIpHandler`.

Then you can tweak the call to it :
```java
        // Add metrics servlet
        webService.addServlet("/metrics",
                new ServletHolder(metricsServlet),
                config.isAuthenticateMetricsEndpoint(), attributeMap);
```
by adding the `isGzip` based on `pulsar.getConfiguration().isCompressOutputMetricsInPrometheus()` 


When that is done, the person looking at the call will have all the information at the end, and we won't duplicate `/metrics` twice

Reply:
Great point @asafm.